ludwig/hyperopt/execution.py

```diff
@@ -696,13 +602,17 @@ def execute(
 
         mode = "min" if self.goal != MAXIMIZE else "max"
         metric = "metric_score"
-        if self.search_alg_dict is not None:
-            if TYPE not in self.search_alg_dict:
+        # if random seed not set, use Ludwig seed
+        self.search_algorithm.check_for_random_seed(random_seed)
+        if self.search_algorithm.search_alg_dict is not None:
+            if TYPE not in self.search_algorithm.search_alg_dict:
                 logger.warning("WARNING: Kindly set type param for search_alg " "to utilize Tune's Search Algorithms.")
```

Comment:
The warning message could be made more useful. We shoudl say that the search alg is not specified, that it should be set from the available ones (keys of the registry?) and which one it is set by default to.